escrow/free_call_api.go

```diff
@@ -0,0 +1,77 @@
+package escrow
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	"math/big"
+
+	"github.com/singnet/snet-daemon/blockchain"
+)
+
+func FreeCallID(userAddress *common.Address, serviceId string) string {
+	return fmt.Sprintf("%v/%v", userAddress, serviceId)
+}
+
+// To Support Free calls
+type FreeCallPayment struct {
+	//Has the Id of the user making the call
+	UserId string
+	//Service Id .
+	ServiceId string
+	//Organization Id
+	OrganizationId string
+	//Current block number
+	CurrentBlockNumber *big.Int
+	// Signature passed
+	Signature []byte
+}
+
+type FreeCallUserKey struct {
+	UserId         string
+	OrganizationId string
+	ServiceId      string
+	GroupID        string
+}
+
+func (key *FreeCallUserKey) String() string {
+	return fmt.Sprintf("{ID:%v/%v/%v/%v}", key.UserId, key.OrganizationId,
+		key.ServiceId, key.GroupID)
+}
+
+type FreeCallUserData struct {
+	UserId        string
+	ServiceId     string
+	OrgId         string
+	GroupID       [32]byte
```

Comment:
Keep GroupID as consistent with ServiceId and OrgId.
Keep the OrgId as consistent naming convention where some places it is mentioned as OrganizationId.